csharp/test/IceSSL/configuration/AllTests.cs

```diff
@@ -1399,73 +1363,39 @@ public static IServerFactoryPrx allTests(TestHelper helper, string testDir)
                     new Communicator(ref args, clientProperties);
                     TestHelper.Assert(false);
                 }
-                catch (System.Exception)
+                catch (Exception)
                 {
                     // Expected.
                 }
             }
             {
-                //
                 // Test password failure with callback.
-                //
                 clientProperties = CreateProperties(defaultProperties, "c_rsa_ca1");
-                clientProperties["Ice.InitPlugins"] = "0";
                 // Don't specify the password.
                 clientProperties.Remove("IceSSL.Password");
-                using var comm = new Communicator(ref args, clientProperties);
-                SslPlugin? plugin = (SslPlugin?)comm.GetPlugin("IceSSL");
-                TestHelper.Assert(plugin != null);
-                var cb = new PasswordCallback("bogus");
-                plugin.SetPasswordCallback(cb);
+
                 try
                 {
-                    comm.InitializePlugins();
+                    using var comm = new Communicator(ref args, clientProperties,
+                        passwordCallback: _ => new NetworkCredential("", "bogus").SecurePassword);
                     TestHelper.Assert(false);
                 }
-                catch (System.Exception)
+                catch (Exception)
                 {
                     // Expected.
                 }
             }
-            {
-                //
-                // Test installation of password callback.
-                //
-                clientProperties = CreateProperties(defaultProperties, "c_rsa_ca1");
-                clientProperties["Ice.InitPlugins"] = "0";
-                // Don't specify the password.
-                clientProperties.Remove("IceSSL.Password");
-                Communicator comm = new Communicator(ref args, clientProperties);
-                SslPlugin? plugin = (SslPlugin?)comm.GetPlugin("IceSSL");
-                TestHelper.Assert(plugin != null);
-                var cb = new PasswordCallback();
-                plugin.SetPasswordCallback(cb);
-                TestHelper.Assert(plugin.GetPasswordCallback() == cb);
-                try
-                {
-                    comm.InitializePlugins();
-                }
-                catch (System.Exception ex)
-                {
-                    Console.WriteLine(ex.ToString());
-                    TestHelper.Assert(false);
-                }
-                comm.Destroy();
-            }
-            {
-                //
+
+            // TODO allow to set the delegate using a property
+            /*{
```

Comment:
I'm not convinced we should add this support for configuring callbacks through properties back.